src/agent/src/device.rs

```diff
@@ -379,6 +379,51 @@ pub async fn wait_for_pci_device(
     Ok(addr)
 }
 
+#[derive(Debug)]
+struct NetPciMatcher {
+    devpath: String,
+}
+
+impl NetPciMatcher {
+    fn new(relpath: &str) -> NetPciMatcher {
+        let root_bus = create_pci_root_bus_path();
+        let path = Path::new(root_bus.as_str()).join(relpath);
+        let path_str = path.to_str().unwrap();
+
+        NetPciMatcher {
+            devpath: String::from(path_str),
+        }
+    }
+}
+
+impl UeventMatcher for NetPciMatcher {
+    fn is_match(&self, uev: &Uevent) -> bool {
+        uev.devpath == self.devpath && uev.subsystem == "net" && !uev.interface.is_empty()
+    }
+}
+
+pub async fn wait_for_net_interface(
+    sandbox: &Arc<Mutex<Sandbox>>,
+    pcipath: &pci::Path,
+) -> Result<()> {
+    let root_bus_sysfs = format!("{}{}", SYSFS_DIR, create_pci_root_bus_path());
+    let sysfs_rel_path = pcipath_to_sysfs(&root_bus_sysfs, pcipath)?;
+
+    let matcher = NetPciMatcher::new(&sysfs_rel_path);
+
+    // Check if the interface is already added in case network is cold-plugged
+    // or the uevent loop is started before network is added.
+    let sys_path = Path::new("/sys").join(matcher.devpath.as_str()).canonicalize()?;
```

Comment:
`mather.devpath` is an absolute path like "/devices/pci0000:00/...", so `join()` will use it to replace `/sys`.